app/code/Magento/Customer/Model/ForgotPassword.php

```diff
@@ -0,0 +1,100 @@
+<?php declare(strict_types=1);
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Customer\Model;
+
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Api\Data\CustomerInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\State\ExpiredException;
+use Magento\Framework\Phrase;
+
+/**
+ * Class ForgotPassword
+ *
+ * @package Magento\Customer\Model
+ * @SuppressWarnings(PHPMD.LongVariable)
+ */
+class ForgotPassword
```

Comment:
Why it names 'Forgot Password' and:
1. Search customer by token
2. Confirm customer 

Can we make this SOLID?

Reply:
Done